src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/entities/Study.java

```diff
@@ -0,0 +1,140 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.metadata.persistence.entities;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.swagger.annotations.ApiModelProperty;
+import org.hibernate.validator.constraints.NotBlank;
+
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Lob;
+import javax.persistence.ManyToMany;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+import java.util.List;
+
+@Entity
+public class Study {
+
+    public enum Type {
+
+        CASE_CONTROL,
+
+        CONTROL_SET,
+
+        CASE_SET,
+
+        COLLECTION,
+
+        TUMOR,
+
+        MATCHED_NORMAL
+
+    }
+
+    public enum Technology {
+
+        GWAS,
+
+        EXOME_SEQUENCING,
+
+        GENOTYPING,
+
+        ARRAY,
+
+        CURATION
+
+    }
+
+    @ApiModelProperty(position = 1, value = "Study auto generated id", required = true, readOnly = true)
+    @JsonProperty(access = JsonProperty.Access.READ_ONLY)
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private long id;
```

Comment:
If we are gonna use the accession as ID, then I would allow passing the id (right now you can pass ID but it is ignored) and remove the GeneratedValue strategy. We could also do this 🙄  -> https://stackoverflow.com/questions/31158509/how-to-generate-custom-id-using-hibernate-while-it-must-be-primary-key-of-table?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

Hibernate allows for custom classes to generate ID. 

Anyway, in that case, yes, with the current design it should be a 255 String index.